homeassistant/components/habitica/calendar.py

```diff
@@ -0,0 +1,261 @@
+
+    async_add_entities(
+        [
+            HabiticaTodosCalendarEntity(coordinator),
+            HabiticaDailiesCalendarEntity(coordinator),
+            HabiticaRemindersCalendarEntity(coordinator),
+        ]
+    )
+
+
+class HabiticaCalendarEntity(HabiticaBase, CalendarEntity):
+    """Base Habitica calendar entity."""
+
+    def __init__(
+        self,
+        coordinator: HabiticaDataUpdateCoordinator,
+    ) -> None:
+        """Initialize calendar entity."""
+        super().__init__(coordinator, self.entity_description)
+
+
+class HabiticaTodosCalendarEntity(HabiticaCalendarEntity):
+    """Habitica todos calendar entity."""
+
+    entity_description = CalendarEntityDescription(
+        key=HabiticaCalendar.TODOS,
+        translation_key=HabiticaCalendar.TODOS,
+    )
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
+
+        events = [
+            CalendarEvent(
+                start=start,
+                end=start,
+                summary=task["text"],
+                description=task["notes"],
+                uid=task["id"],
+            )
+            for task in self.coordinator.data.tasks
+            if task["type"] == HabiticaTaskType.TODO
+            and not task["completed"]
+            and task.get("date")
+            and (start := to_date(task["date"]))
+            and start >= dt_util.now().date()
+        ]
+        events_sorted = sorted(
+            events,
+            key=lambda event: (
+                event.start,
+                self.coordinator.data.user["tasksOrder"]["todos"].index(event.uid),
+            ),
+        )
+
+        return events_sorted[0] if events_sorted else None
+
+    async def async_get_events(
+        self, hass: HomeAssistant, start_date: datetime, end_date: datetime
+    ) -> list[CalendarEvent]:
+        """Return calendar events within a datetime range."""
+
+        return [
+            CalendarEvent(
+                start=start,
+                end=start,
+                summary=task["text"],
+                description=task["notes"],
+                uid=task["id"],
+            )
+            for task in self.coordinator.data.tasks
+            if task["type"] == HabiticaTaskType.TODO
+            and not task["completed"]
+            and task.get("date")
+            and (start := to_date(task["date"]))
+            and (start_date.date() <= start <= end_date.date())
+        ]
+
+
+class HabiticaDailiesCalendarEntity(HabiticaCalendarEntity):
+    """Habitica dailies calendar entity."""
+
+    entity_description = CalendarEntityDescription(
+        key=HabiticaCalendar.DAILIES,
+        translation_key=HabiticaCalendar.DAILIES,
+    )
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
+
+        today = datetime.combine(
+            dt_util.now().date(), time(), dt_util.DEFAULT_TIME_ZONE
```

Comment:
You can just use `dt_util.now()` here and it will give you a local datetime in the current timezone as described here https://developers.home-assistant.io/docs/core/entity/calendar/#get-events

Reply:
But that's exactly what i need, a tz-aware datetime representation of today (starttime 0:00) for rrule to return also events for today.

Reply:
Is that `dt_util.start_of_local_day()`?

Reply:
lol, that method does the exact same thing 😂
(now that I know this exists, I'd preferably use that instead)